### ISSUE TITLE
DDF jsonata Report template - Add ReadMe sheet and some improvements.

### ISSUE THREAD

**Issue by BSnoeijerCD** [cdisc-org/cdisc-rules-engine#1412]
As discussed with @ASL-rmarshall and also mentioned during the sprint review, we propose the following improvements to the output file:
- add the ReadMe sheet as indicated in the attached example.
- add the CDISC Rule ID column to the Issue Summary sheet as well.
- rename column H in the Issue Detail sheet to 'Attributes'

[CDISC_Pilot_Study_v4_Dirty_20251028_comments.xlsx](https://github.com/user-attachments/files/23235622/CDISC_Pilot_Study_v4_Dirty_20251028_comments.xlsx)

**Comment by ASL-rmarshall** [comment]
@gerrycampion Above are the template changes I mentioned.  I also found that using `instanceType` for entity/dataset is not handled consistently for all parts of the report (the Entity column on the Issue Summary sheet was blank even when the Entity column on the Issue Details sheet was populated).

I fixed this locally by changing https://github.com/cdisc-org/cdisc-rules-engine/blob/233c00e8870df0245e33779bb14b01fed3004426/cdisc_rules_engine/utilities/jsonata_processor.py#L57-L89

to assign entity and they use it in both `error_entity` and `validation_error_container`:
```python
            for result in results:
                entity=result.get("entity") or result.get("dataset") or result.get("instanceType")                
                error_entity = ValidationErrorEntity(
                    value=result,
                    dataset=entity,
                    row=result.get("row"),
                    USUBJID=result.get("USUBJID"),
                    SEQ=result.get("SEQ"),
                    entity=entity,
                    instance_id=result.get("instance_id") or result.get("id"),
                    path=result.get("path") or result.get("_path"),
                )
                errors[entity].append(error_entity)
        validation_error_container = [
            ValidationErrorContainer(
                dataset=entity,
                domain=entity,
                targets=rule.get("output_variables"),
                errors=error,
                message=next(iter(rule.get("actions", [])), {})
                .get("params", {})
                .get("message"),
                status=(
                    ExecutionStatus.SUCCESS.value
                    if results
                    else ExecutionStatus.EXECUTION_ERROR.value
                ),
                entity=entity,
            ).to_representation()
            for entity, error in errors.items()
        ]
        return validation_error_container
```
Could you apply something similar?

**Comment by BSnoeijerCD** [comment]
@gerrycampion @ASL-rmarshall : Attached is an example output with the proposed ReadMe sheet.

[CDISC_Pilot_Study_v4_Dirty_20251028_comments.xlsx](https://github.com/user-attachments/files/23306868/CDISC_Pilot_Study_v4_Dirty_20251028_comments.xlsx)
